src/main/java/io/supertokens/webserver/api/passwordless/ConsumeCodeAPI.java

```diff
@@ -108,6 +110,16 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws I
 
             result.addProperty("createdNewUser", consumeCodeResponse.createdNewUser);
             result.add("user", userJson);
+            if (getVersionFromRequest(req).greaterThanOrEqualTo(SemVer.v4_0)) {
+                for (LoginMethod loginMethod : consumeCodeResponse.user.loginMethods) {
+                    if (loginMethod.recipeId.equals(RECIPE_ID.PASSWORDLESS)
+                            && Objects.equals(loginMethod.email, consumeCodeResponse.email)
```

Comment:
Does this work if the pwless user has both email and phoneNumber associated with them (and they only log in using one)?

Reply:
Implemented and added test for this case